src/modules/flow/timestamp/timestamp.c

```diff
@@ -0,0 +1,482 @@
+
+    return send_timestamp(node, port, mdata);
+}
+
+static int
+localtime_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    time_t value;
+    struct tm split_time;
+    struct sol_irange year = { 0, 0, INT32_MAX, 1 };
+    struct sol_irange mon = { 0, 0, 11, 1 };
+    struct sol_irange mday = { 0, 1, 31, 1 };
+    struct sol_irange hour = { 0, 0, 23, 1 };
+    struct sol_irange min = { 0, 0, 59, 1 };
+    struct sol_irange sec = { 0, 0, 59, 1 };
+    struct sol_irange wday = { 0, 0, 6, 1 };
+    struct sol_irange yday = { 0, 0, 365, 1 };
+    int r;
+
+    r = sol_flow_packet_get_timestamp(packet, &value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    tzset();
+    if (!localtime_r(&value, &split_time)) {
+        sol_flow_send_error_packet(node, EINVAL,
+            "Could not convert time.");
+        return 0;
+    }
+
+    year.val = split_time.tm_year + 1900;
+    r = sol_flow_send_irange_packet(node,
+        SOL_FLOW_NODE_TYPE_TIMESTAMP_SPLIT_TIME__OUT__YEAR,
+        &year);
+    SOL_INT_CHECK(r, < 0, r);
+
+    mon.val = split_time.tm_mon + 1;
+    r = sol_flow_send_irange_packet(node,
+        SOL_FLOW_NODE_TYPE_TIMESTAMP_SPLIT_TIME__OUT__MONTH,
+        &mon);
+    SOL_INT_CHECK(r, < 0, r);
+
+    mday.val = split_time.tm_mday;
+    r = sol_flow_send_irange_packet(node,
+        SOL_FLOW_NODE_TYPE_TIMESTAMP_SPLIT_TIME__OUT__MONTH_DAY,
+        &mday);
+    SOL_INT_CHECK(r, < 0, r);
+
+    hour.val = split_time.tm_hour;
+    r = sol_flow_send_irange_packet(node,
+        SOL_FLOW_NODE_TYPE_TIMESTAMP_SPLIT_TIME__OUT__HOUR,
+        &hour);
+    SOL_INT_CHECK(r, < 0, r);
+
+    min.val = split_time.tm_min;
+    r = sol_flow_send_irange_packet(node,
+        SOL_FLOW_NODE_TYPE_TIMESTAMP_SPLIT_TIME__OUT__MINUTE,
+        &min);
+    SOL_INT_CHECK(r, < 0, r);
+
+    sec.val = split_time.tm_sec;
+    r = sol_flow_send_irange_packet(node,
+        SOL_FLOW_NODE_TYPE_TIMESTAMP_SPLIT_TIME__OUT__SECOND,
+        &sec);
+    SOL_INT_CHECK(r, < 0, r);
+
+    wday.val = split_time.tm_wday;
+    r = sol_flow_send_irange_packet(node,
+        SOL_FLOW_NODE_TYPE_TIMESTAMP_SPLIT_TIME__OUT__WEEK_DAY,
+        &wday);
+    SOL_INT_CHECK(r, < 0, r);
+
+    yday.val = split_time.tm_yday;
+    r = sol_flow_send_irange_packet(node,
+        SOL_FLOW_NODE_TYPE_TIMESTAMP_SPLIT_TIME__OUT__YEAR_DAY,
+        &yday);
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (split_time.tm_isdst < 0)
+        SOL_DBG("Daylight saving time information not available.");
+    else {
+        r = sol_flow_send_boolean_packet(node,
+            SOL_FLOW_NODE_TYPE_TIMESTAMP_SPLIT_TIME__OUT__DAYLIGHT_SAVING_TIME,
+            !!split_time.tm_isdst);
+        SOL_INT_CHECK(r, < 0, r);
+    }
+
+    return 0;
+}
+
+struct timestamp_comparison_data {
+    bool (*func) (time_t var0, time_t var1);
+    time_t var0;
+    time_t var1;
+    uint16_t port;
+    bool var0_initialized : 1;
```

Comment:
`func` and `port` should be class, not instance. Do we have a way to set these only once in the node type description?

I saw that in other node types  but forgot to investigate. This should save some memory by reducing structure size